File: lib/models/post.dart
import 'user.dart';

class Post {
  final String title;
  final String description;
  final DateTime timestamp;
  final User author;
  final String tag;

  Post({
    required this.title,
    required this.description,
    required this.timestamp,
    required this.author,
    required this.tag,
  });
}

File: backend/models/post.js
const mongoose = require('mongoose');

const commentSchema = new mongoose.Schema({
  content: String,
  author: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  timestamp: { type: Date, default: Date.now },
});

const postSchema = new mongoose.Schema({
  title: String,
  description: String,
  author: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  timestamp: { type: Date, default: Date.now },
  comments: [commentSchema], 
  tag: String,
});

module.exports = mongoose.model('Post', postSchema);
    